packages/flame_console/lib/src/commands/commands.dart

```diff
@@ -0,0 +1,113 @@
+import 'package:args/args.dart';
+import 'package:flame/components.dart';
+import 'package:flame/game.dart';
+import 'package:flame_console/src/commands/commands.dart';
+import 'package:flame_console/src/commands/pause_command.dart';
+import 'package:flame_console/src/commands/resume_command.dart';
+
+export 'debug_command.dart';
+export 'ls_command.dart';
+export 'remove_command.dart';
+
+abstract class ConsoleCommand<G extends FlameGame> {
+  ArgParser get parser;
+  String get description;
+
+  List<Component> listAllChildren(Component component) {
+    return [
+      for (final child in component.children) ...[
+        child,
+        ...listAllChildren(child),
+      ],
+    ];
+  }
+
+  void onChildMatch(
+    void Function(Component) onChild, {
+    required Component rootComponent,
+    List<String>? ids,
+    List<String>? types,
+    int? limit,
+  }) {
+    final components = listAllChildren(rootComponent);
+
+    var count = 0;
+
+    for (final element in components) {
+      if (limit != null && count >= limit) {
+        break;
+      }
+
+      ids = ids ?? [];
+      types = types ?? [];
```

Comment:
These can be set as default arguments right?
I also thought there was a lint rule against modifying argument variables...

Reply:
This is not handled yet?

Reply:
Missed it! Done now!